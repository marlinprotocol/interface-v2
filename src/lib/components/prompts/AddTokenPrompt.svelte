<script lang="ts">
	import Button from '$lib/atoms/buttons/Button.svelte';
	import { chainConfigStore, chainStore } from '$lib/data-stores/chainProviderStore';
	import { web3WalletStore } from '$lib/data-stores/walletProviderStore';
	import type { Eip1193Provider } from 'ethers';
	import { staticImages } from '../images/staticImages';
	import type { TokenMetadata } from '$lib/types/environmentTypes';
	import { addToast } from '$lib/data-stores/toastStore';

	export let tokenFor: 'POND' | 'MPOND' = 'POND';
	export let label: string;

	async function addTokenToWallet(
		walletProvider: Eip1193Provider,
		tokenData: TokenMetadata | undefined
	) {
		if (!$chainStore.isValidChain) {
			addToast({
				message: 'You are currently on an unsupported network. Please connect to a valid chain',
				variant: 'error'
			});
		} else {
			if (walletProvider) {
				try {
					const added = await walletProvider.request({
						method: 'wallet_watchAsset',
						params: {
							type: 'ERC20',
							options: {
								address: tokenData?.address,
								symbol: tokenData?.currency,
								decimals: tokenData?.decimal,
								image: staticImages.MarlinLogo
							}
						}
					});
					if (added) {
						addToast({ message: `${tokenFor} added to wallet successfully!`, variant: 'success' });
					} else {
						addToast({
							message: 'Uh-oh, there seems to be an issue, please try again',
							variant: 'info'
						});
					}
				} catch (error) {
					console.log(error);
					addToast({
						message: 'There seems to be an error. Please try again in some time',
						variant: 'error'
					});
				}
			}
		}
	}
</script>

<Button
	variant="greyFilled"
	size="small"
	styleClass="font-medium gap-3"
	onclick={() => addTokenToWallet($web3WalletStore[0].provider, $chainConfigStore.tokens[tokenFor])}
>
	<span class="text-2xl font-light">+</span> Add {label}</Button
>
