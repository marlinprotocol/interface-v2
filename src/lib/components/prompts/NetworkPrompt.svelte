<script lang="ts">
	import ContainerCard from '$lib/atoms/cards/ContainerCard.svelte';
	import Text from '$lib/atoms/texts/Text.svelte';
	import ChainSwitcher from '$lib/components/v2/header/sub-components/ChainSwitcher.svelte';
</script>

<ContainerCard>
	<Text variant="h3" text="Unsupported Network" styleClass="text-center" />
	<span class="mb-8 mt-4 px-2 text-center text-[15px] font-medium text-black/50"
		>Please switch to one of the chains in the dropdown to continue.</span
	>
	<div class="flex items-center justify-center">
		<ChainSwitcher />
	</div>
</ContainerCard>
