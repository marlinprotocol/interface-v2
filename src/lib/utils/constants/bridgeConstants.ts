import type { TableModel } from '$lib/types/v2/componentTypes';

export const POND_TO_MPOND_TABLE_HEADER: TableModel['header'][] = [
	{
		title: 'DATE',
		id: 'date',
		sorting: true
	},
	{
		title: 'POND',
		id: 'pond',
		tooltipText: 'The amount of POND which was placed for conversion to MPond.'
	},
	{
		title: 'MPond',
		id: 'mPond',
		tooltipText:
			'The corresponding amount of MPond received as a result of the conversion from POND. Note: 1 Million POND equals 1 MPond.'
	},
	{
		title: 'TX HASH',
		id: 'txnHash',
		tooltipText: 'Transaction hash of the conversion from POND to MPond.'
	}
];

export const MPOND_TO_POND_TABLE_HEADER: TableModel['header'][] = [
	{
		title: 'DATE',
		id: 'date',
		sorting: true
	},
	{
		title: 'TX HASH',
		id: 'txnHash',
		tooltipText: 'Transaction hash of the conversion request from MPond to POND.'
	},
	{
		title: 'MPond',
		id: 'mPond',
		tooltipText: 'The amount of MPond that was placed for conversion to POND.'
	},
	{
		title: 'TOTAL POND',
		id: 'pond',
		tooltipText:
			'The total amount of POND to be received as a result of the conversion from MPond. Note: 1 MPond equals 1 Million POND.'
	},
	{
		title: 'PENDING',
		id: 'pending',
		tooltipText: 'The amount of POND that is currently locked and pending conversion.'
	},
	{
		title: 'IN PROCESS',
		id: 'process',
		tooltipText: 'The amount of POND in process for conversion.'
	},
	{
		title: 'ELIGIBLE',
		id: 'eligible',
		tooltipText: 'The amount of POND eligible for conversion.'
	},
	{
		title: '',
		id: 'convert'
	}
];

export const MPOND_CONVERSION_HISTORY_TABLE_HEADER: TableModel['header'][] = [
	{
		title: 'DATE',
		id: 'date'
	},
	{
		title: 'CONVERTED',
		id: 'pond',
		tooltipText: 'The amount of POND converted.'
	},
	{
		title: 'TX HASH',
		id: 'txnHash',
		tooltipText: 'The transaction hash of the conversion.'
	}
];

export const MPOND_CONVERSION_CYCLE_TABLE_HEADER: TableModel['header'][] = [
	{
		title: 'ELIGIBLE / PENDING',
		id: 'eligible',
		tooltipText: 'The amount of POND that is eligible/pending for conversion.'
	},
	{
		title: 'TIMESTAMP',
		id: 'timestamp',
		tooltipText: 'When POND is eligible for conversion.'
	}
];
