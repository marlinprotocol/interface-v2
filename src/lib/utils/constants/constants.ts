export const GET_OPTIONS = {
	method: 'GET',
	headers: {
		'Content-Type': 'application/json'
	}
};

export const MEMORY_SUFFIX = ' MB';

export const TABLE_ITEMS_PER_PAGE = 10;

export const DEFAULT_CURRENCY_DECIMALS = 18; // we use POND as a default currency which has its lowest unit in 18 decimals
export const DEFAULT_PRECISION = 4;

export const SECONDS_IN_HOUR = 3600;
export const SECONDS_IN_DAY = 86400;
export const SECONDS_IN_HUNDRED_YEARS = 60 * 60 * 24 * 365 * 100; //not accounting for leap years

export const POND_PRECISIONS = 2;
export const MPOND_PRECISIONS = 6;

export const SIDEBAR_DROPDOWN_LINK_IDS = {
	oyster: 'oyster-sidebar-link',
	relay: 'relay-sidebar-link',
	bridge: 'bridge-sidebar-link'
};
