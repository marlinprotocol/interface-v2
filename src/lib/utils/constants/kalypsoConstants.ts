export const DEFAULT_KALYPSO_STAKE_UTILIZATION = 10n ** 18n;
export const DEFAULT_KALYPSO_COMPUTE_UTILIZATION = 10n ** 18n;
