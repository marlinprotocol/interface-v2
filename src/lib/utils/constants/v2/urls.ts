// _LINK: complete url
// _URL: internal app routing url

// general links
const DISCORD_LINK = 'https://discord.gg/invite/heT4sND';
const MARLIN_BLOG_LINK = 'https://blog.marlin.org/';
const MARLIN_RESEARCH_FORUM_LINK = 'https://research.marlin.org/';
const MARLIN_DOCS_LINK = 'https://docs.marlin.org/';

//dashboard page
const HUB_DASHBOARD_URL = '/dashboard';

//bridge
const BRIDGE_URL = '/bridge';
const POND_HISTORY_PAGE_URL = '/bridge/pondToMPondHistory';
const MPOND_HISTORY_PAGE_URL = '/bridge/mPondToPondHistory';
const ETH_ARB_BRIDGE_LINK =
	'https://bridge.arbitrum.io/?destinationChain=arbitrum-one&sourceChain=ethereum';
const BRIDGE_LEARN_LINK = 'https://docs.marlin.org/learn/bridges';

//relay
const RELAY_URL = '/relay';
const RELAY_CLUSTERS_LINK = 'https://arb1.marlin.org/relay/operator';
const RELAY_REGISTRATION_LINK = 'https://arb1.marlin.org/relay/operator/register';
const RELAY_DELEGATION_LINK = 'https://arb1.marlin.org/relay/stash';
const RELAY_RECEIVER_LINK = 'https://receivers.marlin.org/';

//oyster
const OYSTER_URL = '/oyster';
const OYSTER_MARKETPLACE_URL = '/oyster/marketplace';
const OYSTER_OPERATOR_URL = '/oyster/operator';
const OYSTER_INVENTORY_URL = '/oyster/inventory';
const OYSTER_AUDITORS_URL = '/oyster/auditors';
const OYSTER_INVENTORY_HISTORY_URL = '/oyster/inventory/history';
const OYSTER_OPERATOR_JOBS_URL = '/oyster/operator';
const OYSTER_OPERATOR_HISTORY_URL = '/oyster/operator/history';

//kalypso
const KALYPSO_URL = '/kalypso';
const KALYPSO_LINK_1_URL = '/kalypso';

//ecosystem
const ECOSYSTEM_URL = '/ecosystem';

export const ROUTES = {
	DISCORD_LINK,
	MARLIN_BLOG_LINK,
	MARLIN_RESEARCH_FORUM_LINK,
	MARLIN_DOCS_LINK,
	HUB_DASHBOARD_URL,
	BRIDGE_URL,
	POND_HISTORY_PAGE_URL,
	MPOND_HISTORY_PAGE_URL,
	ETH_ARB_BRIDGE_LINK,
	BRIDGE_LEARN_LINK,
	RELAY_URL,
	RELAY_CLUSTERS_LINK,
	RELAY_REGISTRATION_LINK,
	RELAY_DELEGATION_LINK,
	RELAY_RECEIVER_LINK,
	OYSTER_URL,
	OYSTER_MARKETPLACE_URL,
	OYSTER_OPERATOR_URL,
	OYSTER_INVENTORY_URL,
	OYSTER_AUDITORS_URL,
	KALYPSO_URL,
	KALYPSO_LINK_1_URL,
	ECOSYSTEM_URL,
	OYSTER_INVENTORY_HISTORY_URL,
	OYSTER_OPERATOR_JOBS_URL,
	OYSTER_OPERATOR_HISTORY_URL
};
