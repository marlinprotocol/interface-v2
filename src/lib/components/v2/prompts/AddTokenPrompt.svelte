<script lang="ts">
	import Button from '$lib/atoms/v2/buttons/Button.svelte';
	import { chainConfigStore, chainStore } from '$lib/data-stores/chainProviderStore';
	import { web3WalletStore } from '$lib/data-stores/walletProviderStore';
	import type { Eip1193Provider } from 'ethers';
	import type { TokenMetadata } from '$lib/types/environmentTypes';
	import { addToast } from '$lib/data-stores/v2/toastStore';
	import { staticImages } from '$lib/components/images/staticImages';

	export let tokenFor: 'POND' | 'MPOND' = 'POND';
	export let label: string;

	async function addTokenToWallet(
		walletProvider: Eip1193Provider,
		tokenData: TokenMetadata | undefined
	) {
		if (!$chainStore.isValidChain) {
			addToast({
				message: {
					title: 'Unsupported chain',
					description:
						'You are currently on an unsupported network. Please connect to a valid chain'
				},
				variant: 'error'
			});
		} else {
			if (walletProvider) {
				try {
					const added = await walletProvider.request({
						method: 'wallet_watchAsset',
						params: {
							type: 'ERC20',
							options: {
								address: tokenData?.address,
								symbol: tokenData?.currency,
								decimals: tokenData?.decimal,
								image: staticImages.MarlinLogo
							}
						}
					});
					if (added) {
						addToast({
							message: {
								title: 'Success',
								description: `${tokenFor} added to wallet successfully!`
							},
							variant: 'success'
						});
					} else {
						addToast({
							message: {
								title: 'Try again',
								description: 'Uh-oh, there seems to be an issue, please try again'
							},
							variant: 'info'
						});
					}
				} catch (error) {
					console.log(error);
					addToast({
						message: {
							title: 'Error',
							description: 'There seems to be an error. Please try again in some time'
						},
						variant: 'error'
					});
				}
			}
		}
	}
</script>

<Button
	variant="greyFilled"
	size="small"
	styleClass="font-medium gap-3"
	onclick={() => addTokenToWallet($web3WalletStore[0].provider, $chainConfigStore.tokens[tokenFor])}
>
	<span class="text-2xl font-light">+</span> Add {label}</Button
>
