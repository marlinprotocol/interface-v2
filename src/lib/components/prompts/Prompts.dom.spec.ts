import { describe, it, expect } from 'vitest';
import { render, fireEvent } from '@testing-library/svelte';
import NetworkPrompt from './NetworkPrompt.svelte';
import SmallScreePrompt from './SmallScreenPrompt.svelte';
import UnderConstructionPrompt from './UnderConstructionPrompt.svelte';
import { DISCORD_LINK } from '$lib/utils/constants/urls';
import AddTokenPrompt from './AddTokenPrompt.svelte';

describe('NetworkPrompt', () => {
	it("render's the Network Prompt properly", async () => {
		const { container } = render(NetworkPrompt, { props: {} });
		expect(container).toMatchSnapshot();
	});

	it('should display the proper text and contain the ChainSwitcher component', async () => {
		const { container, getByText } = render(NetworkPrompt, { props: {} });

		const heading = getByText('Unsupported Network');
		const instruction = getByText(
			'Please switch to one of the chains in the dropdown to continue.'
		);

		const chainSwitcher = container.querySelector('#chain-dropdown');

		expect(heading).toBeTruthy();
		expect(instruction).toBeTruthy();
		expect(chainSwitcher).toBeTruthy();
	});
});

describe('SmallScreePrompt', () => {
	// Store the original window.innerWidth value to restore it after each test
	const originalInnerWidth = window.innerWidth;
	// Restore the original window.innerWidth value after each test
	afterEach(() => {
		window.innerWidth = originalInnerWidth;
	});

	it('should not show the small screen prompt by default', () => {
		const { queryByText, container } = render(SmallScreePrompt);

		const smallScreenPrompt = queryByText(
			'Please note that this site is not optimized for mobile devices.'
		);
		expect(smallScreenPrompt).toBeNull();
		expect(container).toMatchSnapshot();
	});

	it('should show the small screen prompt when innerWidth is less than 950', async () => {
		// Act by setting "innerWidth" less than "950"
		window.innerWidth = 600;
		window.dispatchEvent(new Event('resize'));

		const { container, queryByText } = render(SmallScreePrompt);

		// Assertion: Check if the prompt appears
		expect(
			queryByText('Please note that this site is not optimized for mobile devices.')
		).toBeTruthy();
		expect(container).toMatchSnapshot();
	});

	it('should hide the small screen prompt when the "Enter Site" button is clicked', async () => {
		window.innerWidth = 600;
		window.dispatchEvent(new Event('resize'));

		const { queryByText, getByText } = render(SmallScreePrompt);

		// Pre-Assertion: Check if the prompt appears
		expect(
			queryByText('Please note that this site is not optimized for mobile devices.')
		).toBeTruthy();

		// Find the button and click it
		const button = getByText('Enter Site');
		await fireEvent.click(button);

		// Assertion: Check if the prompt disappears
		expect(
			queryByText('Please note that this site is not optimized for mobile devices.')
		).toBeNull();
	});

	it('should not show the small screen prompt when innerWidth is greater than 950', async () => {
		window.innerWidth = 960;
		window.dispatchEvent(new Event('resize'));

		const { queryByText } = render(SmallScreePrompt);

		// Assertion: Check if the prompt does not appear
		expect(
			queryByText('Please note that this site is not optimized for mobile devices.')
		).toBeNull();
	});
});

describe('UnderConstructionPrompt', () => {
	it('renders a heading with the correct text', () => {
		const { getByText } = render(UnderConstructionPrompt);
		expect(getByText('UH OH!')).toBeTruthy();
	});

	it('renders the under-construction message with a link to Discord', () => {
		const { getByRole, getByText } = render(UnderConstructionPrompt);
		const discordLinkElement = getByRole('link', { name: 'discord' });
		expect(discordLinkElement.getAttribute('href')).toBe(DISCORD_LINK);
		expect(getByText('FISHY')).toBeTruthy();
	});

	it('should display the strong tag within the paragraph', () => {
		const { getByText } = render(UnderConstructionPrompt);
		expect(getByText('FISHY', { selector: 'strong' })).toBeTruthy();
	});

	it('renders a button that directs the user to the homepage', () => {
		const { getByRole } = render(UnderConstructionPrompt);
		const homeButton = getByRole('button', { name: 'GO BACK HOME' });
		expect(homeButton.closest('a')?.getAttribute('href')).toBe('/');
	});
});

describe('AddTokenPrompt', () => {
	it('renders the AddTokenPrompt properly', () => {
		const { container } = render(AddTokenPrompt);
		expect(container).toMatchSnapshot();
	});

	it('renders the name of the token passed in', async () => {
		const { getByText } = render(AddTokenPrompt, {
			props: {
				label: 'Add POND',
				tokenFor: 'POND'
			}
		});
		expect(getByText('Add POND')).toBeTruthy();
		render(AddTokenPrompt, { props: { label: 'Add MPOND', tokenFor: 'MPOND' } });
		expect(getByText('Add MPOND')).toBeTruthy();
	});

	it('renders the plus symbol', () => {
		const { getByText } = render(AddTokenPrompt);
		expect(getByText('+')).toBeTruthy();
	});
});
