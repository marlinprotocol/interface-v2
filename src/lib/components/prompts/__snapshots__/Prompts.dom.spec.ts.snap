// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NetworkPrompt > render's the Network Prompt properly 1`] = `
<body>
  <onboard-v2
    style="all: initial;"
  />
  <style>
    
    
@font-face {
  font-family: 'Inter';
  font-style:  normal;
  font-weight: 300 600;
  font-display: swap;
  src: url("https://rsms.me/inter/font-files/Inter-roman.var.woff2?v=3.19") format("woff2");
}

  
  </style>
  <div>
    <div
      class="flex h-full w-full items-center justify-center"
    >
      <div
        class="flex h-[460px] w-130 flex-col items-center justify-center rounded-3xl border border-[#D9DADE] bg-[#3840C7]"
      >
        <div
          class="flex h-[84px] w-[84px] items-center justify-center rounded-full bg-white"
        >
          <img
            alt="Info Icon"
            src="/images/information-purple.svg"
          />
        </div>
         
        <span
          class="mt-2 text-3xl font-medium text-[#D9DADE]"
        >
          Unsupported Network
        </span>
         
        <p
          class="text-md mx-auto mt-2 max-w-[400px] text-center font-light leading-6 text-[#D9DADE]"
        >
          Please switch to one of the chains in the dropdown to continue.
        </p>
         
        <div
          class="mt-12 flex w-64 items-center justify-between rounded-full border border-white"
        >
          <span
            class="pl-4 font-medium text-[#D9DADE]"
          >
            Switch
          </span>
           
          <div
            class="dropdown dropdown-end"
            id="chain-dropdown"
          >
            <div
              class="btn btn-theme btn-sm items-center flex justify-center disabled:cursor-not-allowed !rounded-[100px] bg-white text-black border border-[#D9DADE] hover:bg-base-200 chain-btn m-1 flex h-12"
              role="button"
              tabindex="0"
            >
              <div
                class="flex items-center justify-center"
              >
                <div
                  class="flex h-8 w-8 items-center justify-center"
                >
                  <img
                    alt="current chain"
                    src="/images/arbitrum.webp"
                  />
                </div>
                 
                <div
                  class="ml-2"
                >
                  <svg
                    aria-label=""
                    class="fa-icon icon-primary s-T5YyQewRBd7n"
                    data-testid="icon"
                    height="16"
                    role="presentation"
                    style="width:10px; height:10px; cursor:pointer;  "
                    version="1.1"
                    viewBox="0 0 1792 1792"
                    width="16"
                  >
                    <path
                      d="M1683 808l-742 741q-19 19-45 19t-45-19l-742-741q-19-19-19-45.5t19-45.5l166-165q19-19 45-19t45 19l531 531 531-531q19-19 45-19t45 19l166 165q19 19 19 45.5t-19 45.5z"
                    />
                    
                     
                     
                    
                    
                  </svg>
                  <!--&lt;Svg&gt;-->
                  <!--&lt;Icon&gt;-->
                  <!--&lt;Icon&gt;-->
                </div>
              </div>
            </div>
             
            <ul
              class="menu dropdown-content rounded-box z-[1] mt-4 w-52 bg-base-100 p-2 shadow"
              tabindex="0"
            >
              <li
                class="flex rounded-lg bg-primary text-white"
              >
                <button>
                  <div
                    class="h-6 w-6 rounded-full ring-1 ring-white"
                  >
                    <img
                      alt="Chain Logo"
                      src="/images/arbitrum.webp"
                    />
                  </div>
                   
                  Arbitrum Sepolia
                </button>
                 
              </li>
            </ul>
          </div>
          <!--&lt;ChainSwitcher&gt;-->
        </div>
      </div>
    </div>
    <!--&lt;NetworkPrompt&gt;-->
  </div>
</body>
`;
